app/controllers/responsible_body/donated_devices/interest_controller.rb

```diff
@@ -0,0 +1,227 @@
+class ResponsibleBody::DonatedDevices::InterestController < ResponsibleBody::BaseController
+  before_action :redirect_if_already_completed, except: %i[select_schools opted_in]
+  before_action :find_request!, only: %i[select_schools device_types how_many_devices address disclaimer check_answers]
+
+  def new
+    @form = DonatedDeviceInterestForm.new
+  end
+
+  def create
+    @form = DonatedDeviceInterestForm.new(device_interest_params)
+
+    if @form.valid?
+      if @form.interested?
+        redirect_to responsible_body_donated_devices_about_devices_path(@school)
+      else
+        render :not_interested
+      end
+    else
+      render :new, status: :unprocessable_entity
+    end
+  end
+
+  def about; end
+
+  def queue; end
+
+  def interest_confirmation
+    @form = DonatedDeviceInterestForm.new(device_interest_params)
+    if request.post?
+      if @form.valid?
+        if @form.interested?
+          redirect_to responsible_body_donated_devices_all_or_some_schools_path
+        else
+          render :not_interested
+        end
+      else
+        render :interest_confirmation, status: :unprocessable_entity
+      end
+    end
+  end
+
+  def all_or_some_schools
+    @request = find_or_build_request
+    if request.post?
+      last_status = @request.status
+      @request.assign_attributes(donated_device_params.merge(status: 'opt_in_step'))
+      if @request.valid?
+        @request.status = last_status
+        if @request.opt_in_all_schools?
+          # add all schools
+          @request.schools = all_centrally_managed_schools_ids
+          @request.save!
+          if @request.complete?
+            redirect_to responsible_body_donated_devices_opted_in_path
+          else
+            redirect_to responsible_body_donated_devices_what_devices_do_you_want_path
+          end
+        else
+          @request.save!
+          redirect_to responsible_body_donated_devices_select_schools_path
+        end
+      else
+        render :all_or_some_schools, status: :unprocessable_entity
+      end
+    end
+  end
+
+  def select_schools
```

Comment:
Ditto - this method is even longer than the previous one, and some of the logic seems common between the two - could some of the cases be factored out to a separate method?

I'm thinking of simple things like:

```
          if @request.complete?
            redirect_to responsible_body_donated_devices_opted_in_path
          else
            redirect_to responsible_body_donated_devices_what_devices_do_you_want_path
          end
```

could be refactored to:

```
redirect_to next_url_for(@request)

(...)

def next_url_for(request: @request)
  request.complete? responsible_body_donated_devices_opted_in_path : responsible_body_donated_devices_what_devices_do_you_want_path
end
```
etc